modules/afsocket/afsocket-source.c

```diff
@@ -703,6 +813,9 @@ afsocket_sd_init_method(LogPipe *s)
 {
   AFSocketSourceDriver *self = (AFSocketSourceDriver *) s;
 
+  if (self->dynamic_window_ctr)
+    dynamic_window_counter_init(self->dynamic_window_ctr);
```

Comment:
In case of reload, this resets `window` value to `iw_size`, which is going to cause "overflow" when *dynamic* window is released.

Reply:
Nice catch, we've seen this too while running reload tests.